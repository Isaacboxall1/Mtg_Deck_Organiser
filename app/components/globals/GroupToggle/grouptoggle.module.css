.groupToggleContainer {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-around;
    font-family: 'Comfortaa', cursive;
    height: 100px;
    margin: 100px;
}


.groupToggle {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    height: 70px;
    width: 300px;
    border-radius: 25px;
    overflow: hidden;
    transition: all 0.2s ease-in-out;
}

.groupToggle button {
    height: 100%;
    width: 33.33%;
    border: none;
    font-family: 'comfortaa', cursive;
    font-weight: 800;
    font-size: 1.3rem;
    color:rgb(91, 88, 88)
}

.groupToggle .inactive:hover {
    background-color: #e5e2e2;
    cursor: pointer;
}
.active {
    background-color: orange;
}

.groupToggle :nth-child(1) {
    border-right: 1px solid rgb(190, 186, 186);
}

.groupToggle :nth-child(2) {
    border-right: 1px solid rgb(190, 186, 186);
}