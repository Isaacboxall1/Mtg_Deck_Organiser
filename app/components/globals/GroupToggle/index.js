import styles from './grouptoggle.module.css';
export default function GroupToggle({setSortCriteria, sortCriteria}) {

    return ( 
    <div className={styles.groupToggleContainer}>
        <label className={styles.label}>Group By:</label>
    <div className={styles.groupToggle}>
    <button className={sortCriteria === 'type' ? styles.active : styles.inactive} onClick={()=> {setSortCriteria('type')}}>Type</button>
    <button className={sortCriteria === 'color' ? styles.active : styles.inactive} onClick={()=> {setSortCriteria('color')}}>Color</button>
    <button className={sortCriteria === 'rarity' ? styles.active : styles.inactive} onClick={()=> {setSortCriteria('rarity')}}>Rarity</button>
    </div>
    </div>)
}
