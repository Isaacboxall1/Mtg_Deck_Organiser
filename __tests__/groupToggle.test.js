import GroupToggle from "@/app/components/globals/GroupToggle";
import { render } from "@testing-library/react";

const mockProps = {
    sortCriteria: 'type',
    setSortCriteria: jest.fn()
}

describe('GroupToggle', () => {
    test('should render correctly', () => {
        const { container } = render(<GroupToggle />);
        expect(container).toMatchSnapshot();
    }
    )
    test('should render with props', () => {
        const { container } = render(<GroupToggle {...mockProps} />);
        expect(container).toMatchSnapshot();
    }
    )
   test('active class should be applied to button with matching sortCriteria', () => {
        const { getByText } = render(<GroupToggle {...mockProps} />);
        expect(getByText('Type').classList.contains('active')).toBe(true);
        expect(getByText('Color').classList.contains('inactive')).toBe(true);
        expect(getByText('Rarity').classList.contains('inactive')).toBe(true);
    }
    )
    test('setSortCriteria should be called when button is clicked', () => {
        const { getByText } = render(<GroupToggle {...mockProps} />);
        getByText('Color').click();
        expect(mockProps.setSortCriteria).toHaveBeenCalled();
    }
    )
})