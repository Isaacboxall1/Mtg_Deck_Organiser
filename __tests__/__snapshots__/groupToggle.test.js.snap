// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupToggle should render correctly 1`] = `
<div>
  <div
    class="groupToggleContainer"
  >
    <label
      class="label"
    >
      Group By:
    </label>
    <div
      class="groupToggle"
    >
      <button
        class="inactive"
      >
        Type
      </button>
      <button
        class="inactive"
      >
        Color
      </button>
      <button
        class="inactive"
      >
        Rarity
      </button>
    </div>
  </div>
</div>
`;

exports[`GroupToggle should render with props 1`] = `
<div>
  <div
    class="groupToggleContainer"
  >
    <label
      class="label"
    >
      Group By:
    </label>
    <div
      class="groupToggle"
    >
      <button
        class="active"
      >
        Type
      </button>
      <button
        class="inactive"
      >
        Color
      </button>
      <button
        class="inactive"
      >
        Rarity
      </button>
    </div>
  </div>
</div>
`;
